### ISSUE TITLE
Build fails if shared  on macOS due to value of patch version

### ISSUE THREAD

**Issue by SpaceIm** [tlx/tlx#31]
patch version is greater than 1023, the linker of AppleClang doesn't allow such big value:

```
[100%] Linking CXX shared library ../../lib/libtlx.dylib
ld: malformed 64-bit a.b.c.d.e version number: 0.5.20200222
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by SpaceIm** [comment]
@bingmann why did you close this issue as completed? I don't see any fix in master branch or latest release.

**Comment by bingmann** [comment]
Ah, I tested it on alpine/musl and missed this one.

**Comment by bingmann** [comment]
Ah yes, it built on the MacOS CI and I thought it was fine. But yes, this hasn't changed. I'll look into it tomorrow.

**Comment by bingmann** [comment]
Fixed with 0.6.0
